src/main/java/pl/edu/mimuw/chess/Bishop.java

```diff
@@ -0,0 +1,37 @@
+package pl.edu.mimuw.chess;
+
+public class Bishop extends Piece {
+    public Bishop(Board board, Color color) {
+        super(board, color);
+    }
+    public boolean isLegal(Position position) {
+        if(!Position.validPosition(position)) return false;
```

Comment:
dużo powtarzającego się kodu we wszystkich klasach, to jest do wyabstrachowania do pomocniczej funkcji (albo takiej co woła bardziej szcegółową po zrobieniu ogólnych checkow albo pomocniczej)